src/BatchClientImpl.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\Exception;
+use Http\Client\HttpClient;
+use Http\Client\Common\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class BatchClientImpl implements BatchClient
```

Comment:
Maybe `SimpleBatchClient` just to remove the `impl` :/

Reply:
Or `CommonBatchClient`.. same suggestions for `router` also.

Reply:
i like the "simple" prefix

Reply:
we now use the Interface suffix for those